src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -274,14 +272,15 @@ private ExecutionResultNode convertSingleHydrationResultIntoOverallResult(Execut
                                                                               ExecutionStepInfo hydratedFieldStepInfo,
                                                                               HydrationTransformation hydrationTransformation,
                                                                               RootExecutionResultNode rootResultNode,
-                                                                              QueryTransformationResult queryTransformationResult) {
+                                                                              QueryTransformationResult queryTransformationResult,
+                                                                              NadelContext nadelContext) {
 
         synthesizeHydratedParentIfNeeded(fieldTracking, hydratedFieldStepInfo);
 
         Map<String, FieldTransformation> transformationByResultField = queryTransformationResult.getTransformationByResultField();
         Map<String, String> typeRenameMappings = queryTransformationResult.getTypeRenameMappings();
         ExecutionResultNode firstTopLevelResultNode = serviceResultNodesToOverallResult
-                .convertChildren(executionId, forkJoinPool, rootResultNode.getChildren().get(0), overallSchema, hydratedFieldStepInfo, true, false, transformationByResultField, typeRenameMappings);
+                .convertChildren(executionId, forkJoinPool, rootResultNode.getChildren().get(0), overallSchema, hydratedFieldStepInfo, true, false, transformationByResultField, typeRenameMappings, nadelContext);
```

Comment:
I think this would be better if we created a `getFirstAndOnlyChild()` method with assert

This would read better to a consumer